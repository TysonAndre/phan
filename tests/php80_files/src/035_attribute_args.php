<?php
/** @deprecated */
#[Attribute]
class DeprecatedAttribute35 {
    private function __construct() {}
}

#[Attribute(Attribute::TARGET_ALL|Attribute::IS_REPEATABLE)]
class TypedAttribute35 {
    private function __construct(public int $flags) { }
}

#[Attribute([])]
#[DeprecatedAttribute35('extra')]
#[TypedAttribute35([])]
#[TypedAttribute35]
class MyAttribute35 {}

