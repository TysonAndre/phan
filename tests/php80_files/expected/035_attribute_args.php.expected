%s:5 PhanEmptyPrivateMethod Empty private method \DeprecatedAttribute35::__construct()
%s:10 PhanCompatibleConstructorPropertyPromotion Cannot use constructor property promotion before php 8.0 for public int $flags of \TypedAttribute35::__construct(int $flags)
%s:10 PhanWriteOnlyPublicProperty Possibly zero read references to public property \TypedAttribute35->flags
%s:13 PhanTypeMismatchArgumentInternalReal Argument 1 ($flags) is [] of type array{} but \Attribute::__construct() takes int
%s:14 PhanAccessNonPublicAttribute Attempting to access attribute \DeprecatedAttribute35 with non-public constructor \DeprecatedAttribute35::__construct() defined at %s:4. This will throw if ReflectionAttribute->newInstance() is called.
%s:14 PhanDeprecatedClass Using a deprecated class \DeprecatedAttribute35 defined at %s:4
%s:14 PhanParamTooMany Call with 1 arg(s) to \DeprecatedAttribute35::__construct() which only takes 0 arg(s) defined at %s:5
%s:15 PhanAccessNonPublicAttribute Attempting to access attribute \TypedAttribute35 with non-public constructor \TypedAttribute35::__construct() defined at %s:9. This will throw if ReflectionAttribute->newInstance() is called.
%s:15 PhanTypeMismatchArgumentReal Argument 1 ($flags) is [] of type array{} but \TypedAttribute35::__construct() takes int defined at %s:10
%s:16 PhanAccessNonPublicAttribute Attempting to access attribute \TypedAttribute35 with non-public constructor \TypedAttribute35::__construct() defined at %s:9. This will throw if ReflectionAttribute->newInstance() is called.
%s:16 PhanParamTooFew Call with 0 arg(s) to \TypedAttribute35::__construct(int $flags) which requires 1 arg(s) defined at %s:10
%s:17 PhanUnreferencedClass Possibly zero references to class \MyAttribute35
